__tests__/integration/app.test.js

```diff
@@ -179,6 +177,67 @@ describe("GET /api/articles", () => {
 	});
 });
 
+describe("GET /api/articles/:article_id/comments", () => {
+	test("200: responds with an array of comments for the given article_id with correct properties", () => {
+		return request(app)
+			.get("/api/articles/3/comments")
+			.expect(200)
+			.then(({ body }) => {
+				const { comments } = body;
+				expect(comments).toBeInstanceOf(Array);
+				expect(comments).toHaveLength(2);
+				comments.forEach((comment) => {
+					expect(comment).toMatchObject({
+						comment_id: expect.any(Number),
+						votes: expect.any(Number),
+						created_at: expect.any(String),
+						author: expect.any(String),
+						body: expect.any(String),
+						article_id: expect.any(Number),
+					});
+				});
+			});
+	});
+	test("comment objects in the array should be sorted by date in a descending order", () => {
+		return request(app)
+			.get("/api/articles/1/comments")
+			.expect(200)
+			.then(({ body }) => {
+				const { comments } = body;
+
+				expect(comments).toBeSortedBy("created_at", {
+					descending: true,
+				});
+			});
+	});
+	test("400: responds with appropriate message when passed an invalid article_id", () => {
+		return request(app)
+			.get("/api/articles/not-a-valid-id/comments")
+			.expect(400)
+			.then((response) => {
+				expect(response.body.msg).toBe("Bad request");
+			});
+	});
+	test("404: responds with an appropriate error message when pass a valid but non-existent article_id ", () => {
+		return request(app)
+			.get("/api/articles/999/comments")
+			.expect(404)
+			.then((response) => {
+				expect(response.body.msg).toBe("article does not exist");
+			});
+	});
+	test("200: responds with an empty array if article_id is valid but there are no comments", () => {
+		return request(app)
+			.get("/api/articles/2/comments")
+			.expect(200)
+			.then(({ body }) => {
+				const { comments } = body;
+        expect(comments).toBeInstanceOf(Array);
+				expect(comments).toHaveLength(0);
+			});
```

Comment:
Really good test, as an added bonus, having an assertion here which shows that the comments array is an empty array specifically would be a nice to have.